Update change in `kubernetesClient.customResource` in spring-boot-crd Quickstart

### Describe the bug

Running the spring-boot-crd project throws an error of Kubernetes not found.
mvn version: 3.8.6
minikube version: 1.37.1

### Eclipse JKube version

1.10.0

### Component

Kubernetes Maven Plugin

### Apache Maven version

other (please specify in additional context)

### Gradle version

None

### Steps to reproduce

run `mvn spring-boot:run` once you move into the project.

### Expected behavior

The project has to run.

### Runtime

Kubernetes (vanilla), minikube

### Kubernetes API Server version

other (please specify in additional context)

### Environment

macOS

### Eclipse JKube Logs

```shell
ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project spring-boot-crd: Compilation failure
[ERROR] {path-to-project}/jkube/quickstarts/maven/spring-boot-crd/src/main/java/org/eclipse/jkube/quickstart/springboot/crd/FrameworkResource.java:[41,56] cannot find symbol
[ERROR]   symbol:   method customResource(io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext)
[ERROR]   location: variable kubernetesClient of type io.fabric8.kubernetes.client.KubernetesClient
```


### Sample Reproducer Project

_No response_

### Additional context

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -23,6 +23,7 @@ Usage:
 ### 1.14-SNAPSHOT
 * Fix #1713: Add HelidonHealthCheckEnricher to add Kubernetes health checks for Helidon applications
 * Fix #1714: Add HelidonGenerator to add opinionated container image for Helidon applications
+* Fix #1985: Update outdated methods in Spring Boot CRD Maven Quickstart
 
 ### 1.13.1 (2023-06-16)
 * Fix #2212: Bump Kubernetes Client version to 6.7.2 (fixes issues when trace-logging OpenShift builds -regression in 6.7.1-)

--- a/quickstarts/maven/spring-boot-crd/src/main/java/org/eclipse/jkube/quickstart/springboot/crd/Application.java
+++ b/quickstarts/maven/spring-boot-crd/src/main/java/org/eclipse/jkube/quickstart/springboot/crd/Application.java
@@ -13,8 +13,8 @@
  */
 package org.eclipse.jkube.quickstart.springboot.crd;
 
-import io.fabric8.kubernetes.client.DefaultKubernetesClient;
 import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
 import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
@@ -29,7 +29,7 @@ public class Application {
 
   @Bean
   public KubernetesClient kubernetesClient() {
-    return new DefaultKubernetesClient();
+    return new KubernetesClientBuilder().build();
   }
 
   @Bean

--- a/quickstarts/maven/spring-boot-crd/src/main/java/org/eclipse/jkube/quickstart/springboot/crd/FrameworkResource.java
+++ b/quickstarts/maven/spring-boot-crd/src/main/java/org/eclipse/jkube/quickstart/springboot/crd/FrameworkResource.java
@@ -13,6 +13,8 @@
  */
 package org.eclipse.jkube.quickstart.springboot.crd;
 
+import io.fabric8.kubernetes.api.model.GenericKubernetesResource;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
 import org.springframework.beans.factory.annotation.Autowired;
@@ -20,7 +22,6 @@ import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 import java.util.List;
-import java.util.Map;
 import java.util.stream.Collectors;
 
 @RestController
@@ -35,10 +36,11 @@ public class FrameworkResource {
     this.crdContext = crdContext;
   }
 
-  @SuppressWarnings("unchecked")
   @GetMapping
   public List<String> get() {
-    return ((List<Map<String, Object>>)kubernetesClient.customResource(crdContext).list().get("items")).stream()
-        .map(m -> (String)m.get("name")).collect(Collectors.toList());
+    return (kubernetesClient.genericKubernetesResources(crdContext).list().getItems()).stream()
+        .map(GenericKubernetesResource::getMetadata)
+        .map(ObjectMeta::getName)
+        .collect(Collectors.toList());
   }
 }

